nmigen/vendor/xilinx_spartan_3_6.py

```diff
@@ -126,12 +135,28 @@ class XilinxSpartan6Platform(TemplatedPlatform):
         """,
         r"""
         {{get_tool("bitgen")}}
-            {{get_override("bitgen_opts")|default(["-w"])|options}}
+            {{get_override("bitgen_opts")|default(["-g Binary:Yes -w"])|options}}
```

Comment:
Should this be a defaulted option? This means that anyone who provides `NMIGEN_bitgen_opts` for a board that requires `bitstream.bin` will break the programmer for that board unless they specifically know the default. That seems bad. What are the drawbacks of `-g Binary:Yes`?

Reply:
>What are the drawbacks of -g Binary:Yes

It generates `bitstream.bin` _in addition_ to `bitstream.bit`. I don't see any drawbacks besides the one you mentioned. 

In my case, Mercury _needs_ `-g Binary:Yes` unconditionally- the programmer for that board cannot handle `bitstream.bit`. I suppose I could do an override of `command_templates` at the platform level to add in the `-g Binary:Yes` option.

Reply:
Why not just generate `bitstream.bin` on every board? Unless it's extremely slow or something.

Reply:
>Why not just generate bitstream.bin on every board? Unless it's extremely slow or something.

That is exactly what `-g Binary:Yes` already does. The issue, AIUI, is an override would disable this option unless the user already knew that the option `-g Binary:Yes` is required to get `.bin` files from `bitgen`, correct?

Reply:
Yes. If boards rely on it it shouldn't be an override. It should either be a per-board switch based on a property, or always enabled. If it's not too slow, always enabled is preferable.